dev-util/rebar/rebar-3.14.4.ebuild

```diff
@@ -0,0 +1,101 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1
+
+HEX_PACKAGES="
+bbmustache-1.10.0
+certifi-2.5.3
+cf-0.3.1
+cth_readable-1.5.0
+erlware_commons-1.4.0
+eunit_formatters-0.5.0
+getopt-1.0.1
+meck-0.8.13
+parse_trans-3.3.1
+providers-1.8.1
+relx-4.3.0
+ssl_verify_fun-1.1.6
+"
+
+S="${WORKDIR}/rebar3-${PV}"
+
+# TODO: Move to eclass.
+# @FUNCTION: hex_package_uris
+# @DESCRIPTION:
+# Generates the URIs to put in SRC_URI to help fetch dependencies from hex.pm.
+hex_package_uris() {
+	local package
+	for package in "$@"; do
+		echo "https://repo.hex.pm/tarballs/$package.tar -> $package.hex.tar"
+	done
+}
+
+# Tests currently failing.
+RESTRICT="test"
+
+DESCRIPTION="Erlang build tool that makes it easy to build and test applications and releases"
+HOMEPAGE="http://www.rebar3.org/ https://github.com/erlang/rebar3/"
+SRC_URI="https://github.com/erlang/rebar3/archive/${PV}.tar.gz -> ${P}.tar.gz
+	$(hex_package_uris ${HEX_PACKAGES})"
+
+LICENSE="Apache-2.0 MIT BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~arm64 ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x64-solaris"
+IUSE=""
+
+RDEPEND="
+	dev-lang/erlang:=
+	!dev-util/rebar-bin
+"
+DEPEND="${RDEPEND}"
+
+src_unpack() {
+	local archive
+	for archive in ${A}; do
+		case "${archive}" in
+			*.hex.tar)
+				ebegin "Unpacking ${archive}"
+
+				local dest
+
+				# TODO: Create an array of test deps?
+				if [[ "${archive%-*}" == "meck" ]]; then
+					dest="${S}/_build/test/lib/${archive%-*}"
+				else
+					dest="${S}/_build/default/lib/${archive%-*}"
+				fi
+
+				mkdir -p "$dest"
+				tar -O -xf "${DISTDIR}/${archive}" contents.tar.gz |
+					tar -xzf - -C "${dest}" || die
+
+				eend $?
+				;;
+			*)
+				unpack "${archive}"
+				;;
+		esac
+	done
+
+	mkdir -p "${T}/rebar_cache_dir/hex" || die
+	xzcat "${FILESDIR}/packages.idx.xz" > \
```

Comment:
This file comes from running `rebar3 compile && rebar3 ct` outside of the network sandbox so that it can fetch the dependencies from hex.pm. It would be nice to be able to generate this file from the `HEX_PACKAGES` variable somehow.

Also it needs to be compressed, otherwise it's too big and repoman gets upset.

Reply:
Moved to SRC_URI as mentioned on IRC.